.github/workflows/release_build.yaml

```diff
@@ -70,56 +106,69 @@ jobs:
           name: ltheory-bin-ubuntu2004
           path: ${{github.workspace}}/bin/*
 
-  release:
-    name: Update 'latest' release
-    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+  tag:
+    name: New tag
     needs:
+      - gen-build-version
       - build-windows
       - build-macos
       - build-ubuntu
     runs-on: ubuntu-latest
     steps:
-      - name: Download artifacts
-        uses: actions/download-artifact@v3
-      - name: Inspect directory after downloading artifacts
-        run: ls -alFR
-      - name: Zip artifacts
-        run: for dir in $(ls); do cd $dir && zip ../$dir.zip ./* -r && cd ..; done
       - name: Get latest release ID
         uses: octokit/request-action@v2.x
         id: get_latest_release
         with:
-          route: GET /repos/Limit-Theory-Redux/ltheory/releases/tags/latest
+          route: GET /repos/${{github.repository}}/releases/tags/latest
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Delete latest release
+      - name: Delete 'latest' tag ref
         uses: octokit/request-action@v2.x
         with:
-          route: DELETE /repos/Limit-Theory-Redux/ltheory/releases/{release_id}
-          release_id: ${{ fromJson(steps.get_latest_release.outputs.data).id }}
+          route: DELETE /repos/${{github.repository}}/git/refs/tags/latest
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Delete 'latest' tag
+      - name: Delete latest release
         uses: octokit/request-action@v2.x
         with:
-          route: DELETE /repos/Limit-Theory-Redux/ltheory/git/refs/tags/latest
+          route: DELETE /repos/${{github.repository}}/releases/{release_id}
+          release_id: ${{ fromJson(steps.get_latest_release.outputs.data).id }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - name: Re-create 'latest' tag with the current SHA
-        uses: octokit/request-action@v2.x
+      - uses: actions/checkout@v4
         with:
-          route: POST /repos/Limit-Theory-Redux/ltheory/git/refs
-          ref: refs/tags/latest
-          sha: ${{ github.sha }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          fetch-depth: 0
+      - name: Create tag
+        uses: rickstaa/action-create-tag@v1
+        id: "tag_create"
+        with:
+          tag: "latest"
+          force_push_tag: true
+          tag_exists_error: false
+          message: "Build ${{ needs.gen-build-version.outputs.build }}"
+
+  release:
+    name: New release
+    needs:
+      - gen-build-version
+      - tag
```

Comment:
How come you split the tag and release into separate jobs? They feel intrinsically linked to me.

Reply:
Ah i think i did it cause that github-create-tag action needed a checkout and then the script etc folders would be uploaded to the release, i can switch it back to the old way of tagging - do you know how to provide a message over in that one tho?

Reply:
Ah I understand now, because the tag is created locally with a clone right? The old way used the GitHub API to create the tag instead.

To create a named tag, you need to send 2 API requests, one to create the tag object with the message, then another to create the tag that points to the tag object.

The first API call to create the tag object with the message is here: https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28

The second API call would be the `POST /repos/Limit-Theory-Redux/ltheory/git/refs` that used to be here.

A good summary of it is here: https://stackoverflow.com/a/15678196

Reply:
It´s alright, we don´t necessarily need the message right now, we can still add it later with some automatic changelogs